examples/android/clientcache/app/src/main/java/io/grpc/clientcacheexample/ClientCacheExampleActivity.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2015, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.clientcacheexample;
+
+import android.content.Context;
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.text.TextUtils;
+import android.text.method.ScrollingMovementMethod;
+import android.view.View;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.Button;
+import android.widget.CheckBox;
+import android.widget.EditText;
+import android.widget.TextView;
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientInterceptors;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.MethodDescriptor;
+import io.grpc.examples.helloworld.GreeterGrpc;
+import io.grpc.examples.helloworld.HelloReply;
+import io.grpc.examples.helloworld.HelloRequest;
+import io.grpc.stub.ClientCalls;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.concurrent.TimeUnit;
+
+public class ClientCacheExampleActivity extends AppCompatActivity {
+  private static final int CACHE_SIZE_IN_BYTES = 1 * 1024 * 1024; // 1MB
+  private Button mSendButton;
+  private EditText mHostEdit;
+  private EditText mPortEdit;
+  private EditText mMessageEdit;
+  private TextView mResultText;
+  private CheckBox getCheckBox;
+  private CheckBox noCacheCheckBox;
+  private CheckBox onlyIfCachedCheckBox;
+  private ManagedChannel mChannel;
+  private SafeMethodCachingInterceptor.Cache cache;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_clientcacheexample);
+    mSendButton = (Button) findViewById(R.id.send_button);
+    mHostEdit = (EditText) findViewById(R.id.host_edit_text);
+    mPortEdit = (EditText) findViewById(R.id.port_edit_text);
+    mMessageEdit = (EditText) findViewById(R.id.message_edit_text);
+    getCheckBox = (CheckBox) findViewById(R.id.get_checkbox);
+    noCacheCheckBox = (CheckBox) findViewById(R.id.no_cache_checkbox);
+    onlyIfCachedCheckBox = (CheckBox) findViewById(R.id.only_if_cached_checkbox);
+    mResultText = (TextView) findViewById(R.id.grpc_response_text);
+    mResultText.setMovementMethod(new ScrollingMovementMethod());
+
+    cache = SafeMethodCachingInterceptor.newLruCache(CACHE_SIZE_IN_BYTES);
+  }
+
+  public void sendMessage(View view) {
```

Comment:
should this be public?  Is so, javadoc?

Reply:
It needs to be public (https://developer.android.com/guide/topics/ui/controls/button.html#HandlingEvents), Javadoc added.